vowpalwabbit/core/include/vw/core/reductions/epsilon_decay.h

```diff
@@ -22,47 +23,50 @@ namespace epsilon_decay
 struct epsilon_decay_score : scored_config
 {
   epsilon_decay_score() = default;
-  epsilon_decay_score(double alpha, double tau, uint64_t model_idx)
-      : VW::scored_config(alpha, tau), _model_idx(model_idx)
-  {
-  }
+  epsilon_decay_score(double alpha, double tau) : VW::scored_config(alpha, tau) {}
   float decayed_epsilon(uint64_t update_count);
-  uint64_t get_model_idx() const { return _model_idx; }
-
-  uint64_t _model_idx;
 };
 
 struct epsilon_decay_data
 {
-  epsilon_decay_data(uint64_t num_configs, uint64_t min_scope, double epsilon_decay_alpha, double epsilon_decay_tau,
-      parameters& weights, VW::io::logger logger, bool log_champ_changes)
+  epsilon_decay_data(uint64_t model_count, uint64_t min_scope, double epsilon_decay_alpha, double epsilon_decay_tau,
+      dense_parameters& weights, VW::io::logger logger, bool log_champ_changes, bool constant_epsilon)
       : _min_scope(min_scope)
       , _epsilon_decay_alpha(epsilon_decay_alpha)
       , _epsilon_decay_tau(epsilon_decay_tau)
       , _weights(weights)
       , _logger(std::move(logger))
       , _log_champ_changes(log_champ_changes)
+      , _constant_epsilon(constant_epsilon)
   {
-    uint64_t model_idx = 0;
-    for (uint64_t i = 0; i < num_configs; ++i)
+    _scored_configs.reserve(model_count);
+    _weight_indices.resize(model_count);
+    std::iota(_weight_indices.begin(), _weight_indices.end(), 0);
+    for (uint64_t i = 0; i < model_count; ++i)
     {
-      std::vector<epsilon_decay_score> score_vec;
+      _scored_configs.emplace_back(i + 1);
```

Comment:
add a comment about what the `i+1` here is doing, since its not immediately obvious